plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSessionProperties.java

```diff
@@ -387,6 +394,27 @@ public HiveSessionProperties(
                         "Require filter on partition column",
                         hiveConfig.isQueryPartitionFilterRequired(),
                         false),
+                new PropertyMetadata<>(
+                        QUERY_PARTITION_FILTER_REQUIRED_SCHEMAS,
+                        "List of schemas for which filter on partition column is enforced.",
+                        new ArrayType(VARCHAR),
+                        Set.class,
+                        ImmutableSet.of(),
+                        false,
+                        object -> {
+                            if (object == null) {
+                                return ImmutableSet.of();
```

Comment:
i was wrong, you currently cannot have `null` session value:
```
trino:tpch> set session hive.query_partition_filter_required_schemas = null;
Query 20210903_144055_00051_vz757 failed: Unable to set session property 'hive.query_partition_filter_required_schemas' to 'null': Session property value must not be null
io.trino.spi.TrinoException: Unable to set session property 'hive.query_partition_filter_required_schemas' to 'null': Session property value must not be null
	at io.trino.execution.SetSessionTask.execute(SetSessionTask.java:92)
	at io.trino.execution.SetSessionTask.execute(SetSessionTask.java:42)
	at io.trino.execution.DataDefinitionExecution.start(DataDefinitionExecution.java:170)
	at io.trino.execution.SqlQueryManager.createQuery(SqlQueryManager.java:237)
	at io.trino.dispatcher.LocalDispatchQuery.lambda$startExecution$7(LocalDispatchQuery.java:143)
	at io.trino.$gen.Trino_testversion____20210903_130942_3.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: io.trino.spi.TrinoException: Session property value must not be null
	at io.trino.metadata.SessionPropertyManager.evaluatePropertyValue(SessionPropertyManager.java:255)
	at io.trino.execution.SetSessionTask.execute(SetSessionTask.java:87)
	... 8 more
```

this also means `if (object == null)` is a dead code.